core/trino-main/src/test/java/io/trino/sql/query/TestFilterInaccessibleColumns.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.query;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.FeaturesConfig;
+import io.trino.Session;
+import io.trino.metadata.QualifiedObjectName;
+import io.trino.plugin.tpch.TpchConnectorFactory;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.ViewExpression;
+import io.trino.testing.LocalQueryRunner;
+import io.trino.testing.TestingAccessControlManager;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.trino.plugin.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.trino.testing.TestingAccessControlManager.TestingPrivilegeType.SELECT_COLUMN;
+import static io.trino.testing.TestingAccessControlManager.privilege;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@Test(singleThreaded = true) // shared access control
+public class TestFilterInaccessibleColumns
+{
+    private static final String CATALOG = "local";
+    private static final String USER = "user";
+    private static final String SUPER_USER = "super-user";
+
+    private static final Session SESSION = testSessionBuilder()
+            .setCatalog(CATALOG)
+            .setSchema(TINY_SCHEMA_NAME)
+            .setIdentity(Identity.forUser(USER).build())
+            .build();
+
+    private QueryAssertions assertions;
+    private TestingAccessControlManager accessControl;
+
+    @BeforeClass
+    public void init()
+    {
+        LocalQueryRunner runner = LocalQueryRunner.builder(SESSION)
+                .withFeaturesConfig(new FeaturesConfig().setHideInaccesibleColumns(true))
+                .build();
+
+        runner.createCatalog(CATALOG, new TpchConnectorFactory(1), ImmutableMap.of());
+        assertions = new QueryAssertions(runner);
+        accessControl = assertions.getQueryRunner().getAccessControl();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
+    {
+        assertions.close();
+        assertions = null;
+    }
+
+    @Test
+    public void testSimpleTableSchemaFilter()
+    {
+        // No filtering baseline
```

Comment:
Extract testing baselines to separate tests. It is better to have more simple tests, than less complex ones.

Reply:
ok

Reply:
comment is not needed

Reply:
ping to remove the comment